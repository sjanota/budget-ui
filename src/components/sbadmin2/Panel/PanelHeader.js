import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';

export default function PanelHeader({ className, children }) {
  return <div className={classnames(className, 'card-header')}>{children}</div>;
}

PanelHeader.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};
