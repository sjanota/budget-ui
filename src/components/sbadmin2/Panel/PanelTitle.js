import React from 'react';
import PropTypes from 'prop-types';
import { withDictionary } from '../language';

function PanelTitle({ title }) {
  return <h6 className="m-0 font-weight-bold text-primary">{title}</h6>;
}

PanelTitle.propTypes = {
  title: PropTypes.string.isRequired,
};

export default withDictionary('title', PanelTitle);
