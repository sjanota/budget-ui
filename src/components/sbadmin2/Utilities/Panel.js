import React from 'react';
import classnames from 'classnames';
import { withDictionary } from '../language';

export function Panel({ children, className }) {
  return (
    <div className={classnames('card', 'shadow', 'mb-4', className)}>
      {children}
    </div>
  );
}

Panel.Title = function({ title }) {
  return <h6 className="m-0 font-weight-bold text-primary">{title}</h6>;
};

Panel.Title.Dict = withDictionary('title', Panel.Title);

Panel.TitlewithButtons = function({ title, children }) {
  return (
    <div className="d-flex justify-content-between align-items-center">
      <Panel.Title title={title} />
      <div>{children}</div>
    </div>
  );
};

Panel.Header = function({ className, children }) {
  return <div className={classnames(className, 'card-header')}>{children}</div>;
};

Panel.Body = function({ children, className }) {
  return <div className={classnames(className, 'card-body')}>{children}</div>;
};
