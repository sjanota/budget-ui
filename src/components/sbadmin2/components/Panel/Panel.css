.panel-body {
  overflow-y: scroll;
}
