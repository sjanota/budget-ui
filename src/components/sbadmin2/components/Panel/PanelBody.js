import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

export default function PanelBody({ className, children }) {
  return (
    <div className={classnames(className, 'card-body', 'panel-body')}>
      {children}
    </div>
  );
}

PanelBody.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};
