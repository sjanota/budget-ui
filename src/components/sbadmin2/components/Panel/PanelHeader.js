import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';

export default function PanelHeader({ className, ...props }) {
  return <div className={classnames(className, 'card-header')} {...props} />;
}

PanelHeader.propTypes = {
  className: PropTypes.string,
};
