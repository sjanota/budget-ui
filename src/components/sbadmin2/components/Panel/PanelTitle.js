import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { withDictionary } from '../../language';

function PanelTitle({ children, title, className }) {
  const classNames = classnames(
    'm-0',
    'font-weight-bold',
    'text-primary',
    className
  );
  return (
    <h6 className={classNames}>
      {children}
      {title}
    </h6>
  );
}

PanelTitle.propTypes = {
  title: PropTypes.node.isRequired,
  className: PropTypes.string,
  children: PropTypes.node,
};

export default withDictionary('title', PanelTitle);
