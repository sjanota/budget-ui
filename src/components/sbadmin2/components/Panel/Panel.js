import './Panel.css';

import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import PanelBody from './PanelBody';
import PanelHeader from './PanelHeader';
import PanelTitle from './PanelTitle';

export default function Panel({ children, className }) {
  return (
    <div
      className={classnames(
        'card',
        'shadow',
        'mb-4',
        className,
        'overflow-hidden'
      )}
    >
      {children}
    </div>
  );
}

Panel.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};

Panel.Title = PanelTitle;
Panel.Header = PanelHeader;
Panel.Body = PanelBody;
