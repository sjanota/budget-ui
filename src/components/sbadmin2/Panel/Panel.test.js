import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Panel from './Panel';
import PanelHeader from './PanelHeader';
import PanelTitle from './PanelTitle';
import PanelBody from './PanelBody';

describe('Panel', () => {
  it('render with minimal props', () => {
    const component = renderer.create(<Panel />);
    expect(component).toMatchSnapshot();
  });

  it('renders its children', () => {
    const component = mount(
      <Panel>
        <div id="my-div" />
      </Panel>
    );
    expect(component.find('#my-div')).toExist();
  });

  it('contains references to subcomponents', () => {
    const component = mount(
      <Panel>
        <Panel.Title title="" />
        <Panel.Header />
        <Panel.Body />
      </Panel>
    );
    expect(component.find(PanelHeader)).toExist();
    expect(component.find(PanelTitle)).toExist();
    expect(component.find(PanelBody)).toExist();
  });
});
