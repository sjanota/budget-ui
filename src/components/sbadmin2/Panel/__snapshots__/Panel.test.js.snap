// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel render with minimal props 1`] = `
<div
  className="card shadow mb-4"
/>
`;
